crates/orchestrator/src/workers/update_state.rs

```diff
@@ -1,136 +1,153 @@
             Some(job) => {
-                tracing::debug!(job_id = %job.id, "Found latest successful state transition job");
-                let successful_da_jobs_without_successor = config
-                    .database()
-                    .get_jobs_without_successor(JobType::DataSubmission, JobStatus::Completed, JobType::StateTransition)
-                    .await?;
-
-                if successful_da_jobs_without_successor.is_empty() {
-                    tracing::debug!("No new data submission jobs to process");
+                if job.status != JobStatus::Completed {
+                    log::warn!(
+                        "There's already a pending update state job. Parallel jobs can cause nonce issues or can \
+                         completely fail as the update logic needs to be strictly ordered. Returning safely..."
+                    );
                     return Ok(());
                 }
 
-                tracing::debug!(
-                    count = successful_da_jobs_without_successor.len(),
-                    "Found data submission jobs without state transition"
-                );
-
-                let mut metadata = job.metadata;
-                let blocks_to_settle =
-                    Self::parse_job_items_into_block_number_list(successful_da_jobs_without_successor.clone());
-                metadata.insert(JOB_METADATA_STATE_UPDATE_BLOCKS_TO_SETTLE_KEY.to_string(), blocks_to_settle.clone());
-
-                tracing::trace!(blocks_to_settle = %blocks_to_settle, "Prepared blocks to settle for state transition");
-
-                // Creating a single job for all the pending blocks.
-                let new_job_id = successful_da_jobs_without_successor[0].internal_id.clone();
-                match create_job(JobType::StateTransition, new_job_id.clone(), metadata, config).await {
-                    Ok(_) => tracing::info!(job_id = %new_job_id, "Successfully created new state transition job"),
-                    Err(e) => {
-                        tracing::error!(job_id = %new_job_id, error = %e, "Failed to create new state transition job");
-                        return Err(e.into());
-                    }
-                }
-
-                tracing::trace!(log_type = "completed", category = "UpdateStateWorker", "UpdateStateWorker completed.");
-                Ok(())
+                let mut blocks_processed_in_last_job: Vec<u64> = job
+                    .metadata
+                    .get(JOB_METADATA_STATE_UPDATE_BLOCKS_TO_SETTLE_KEY)
+                    .unwrap()
+                    .split(',')
+                    .filter_map(|s| s.parse().ok())
+                    .collect();
+
+                // ideally it's already sorted, but just to be safe
+                blocks_processed_in_last_job.sort();
+
+                let last_block_processed_in_last_job =
+                    blocks_processed_in_last_job[blocks_processed_in_last_job.len() - 1];
```

Comment:
can there be case when we have `JOB_METADATA_STATE_UPDATE_BLOCKS_TO_SETTLE_KEY` empty?